### PR TITLE
Power Platform - Add unit test

### PR DESCRIPTION
**Pull request by andersgMSFT**
Add unit test for Power Platform actions + clean up actions to align with the guidelines we have

### CODE REVIEW

File: Actions/ReadPowerPlatformSettings/ReadPowerPlatformSettings.ps1 [hunk 1]
```diff
@@ -5,22 +5,31 @@ param(
...
 foreach($property in 'ppEnvironmentUrl','companyId','environmentName') {
     if ($deploymentSettings."$property") {
         Write-Host "Setting $property"
         Add-Content -Encoding utf8 -Path $env:GITHUB_OUTPUT -Value "$property=$($deploymentSettings."$property")"
     }
     else {
-        Write-Host "::ERROR::DeployTo$envName setting must contain '$property' property"
-        exit 1
+        # Write-Host "::ERROR::DeployTo$envName setting must contain '$property' property"
+        throw "::ERROR::$envName setting must contain '$property' property"
```
Comment (freddydk) [inline Actions/ReadPowerPlatformSettings/ReadPowerPlatformSettings.ps1]: don't include ::error:: when you throw - that gets added by the try catch handler in action.yaml

File: Actions/ReadPowerPlatformSettings/ReadPowerPlatformSettings.ps1 [hunk 2]
```diff
@@ -5,22 +5,31 @@ param(
...
     }
 }
 
+# Verify the secrets are provided
+if ($null -eq $env:Secrets) {
```
Comment (freddydk) [inline Actions/ReadPowerPlatformSettings/ReadPowerPlatformSettings.ps1]: We don't test this in other places.
But you should update README.md in all actions to reflect expected environment variables and parameters

File: Actions/ReadPowerPlatformSettings/ReadPowerPlatformSettings.ps1 [hunk 3]
```diff
@@ -45,10 +54,14 @@ foreach($secretName in "$($envName)-AuthContext","$($envName)_AuthContext","Auth
             Write-Host "Authenticating with user name"
         }
         else {
-            Write-Host "::ERROR::Secret $secretName must contain either 'ppUserName' and 'ppPassword' properties or 'ppApplicationId', 'ppClientSecret' and 'ppTenantId' properties"
-            exit 1
+            throw "::ERROR::Secret $secretName must contain either 'ppUserName' and 'ppPassword' properties or 'ppApplicationId', 'ppClientSecret' and 'ppTenantId' properties"
```
Comment (freddydk) [inline Actions/ReadPowerPlatformSettings/ReadPowerPlatformSettings.ps1]: remove ::error::

File: Actions/BuildPowerPlatform/BuildPowerPlatform.ps1 [hunk 4]
```diff
@@ -53,7 +56,7 @@ function Update-PowerAppFiles {
             if (Select-String -Pattern $oldSetting -InputObject $fileContent) {
                 $fileContent = $fileContent -creplace $oldSetting, $newSetting
                 Set-Content -Path $file.FullName -Value $fileContent
-                Write-Host "Updated: $file.FullName"
+                Write-Host "Updated: "$file.FullName
```
Comment (freddydk) [inline Actions/BuildPowerPlatform/BuildPowerPlatform.ps1]: Should be "Updated: $($file.FullName)"

File: Tests/ReadPowerPlatformSettings.Action.Test.ps1 [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+            return $errorObject.TargetObject
+        }
+
+        # Test missing ppEnvironmentUrl
+        $deployToDevProperties = @{
+            environmentName = "sandbox"
+            companyId       = "11111111-1111-1111-1111-111111111111"
+        }
+        $errorMessage = runMissingSettingsTest -deployToDevProperties $deployToDevProperties
+        $errorMessage | Should -Be "::ERROR::DeployToDev setting must contain 'ppEnvironmentUrl' property"
```
Comment (freddydk) [inline Tests/ReadPowerPlatformSettings.Action.Test.ps1]: error message shouldn't contain ::ERROR:: (when you remove it from the action)

**Review by freddydk — CHANGES_REQUESTED** [review]

File: Actions/BuildPowerPlatform/BuildPowerPlatform.ps1 [hunk 6]
```diff
@@ -1,9 +1,13 @@
...
-    [Parameter(Position = 1, mandatory = $true)] [string] $EnvironmentName,
-    [Parameter(Position = 2, mandatory = $true)] [string] $SolutionFolder
+    [Parameter(Position = 0, mandatory = $true)] [string] $solutionFolder,
+    [Parameter(Position = 1, mandatory = $false)] [string] $CompanyId,
+    [Parameter(Position = 2, mandatory = $false)] [string] $EnvironmentName,
+    [Parameter(Position = 3, mandatory = $false)] [string] $appBuild,
+    [Parameter(Position = 4, mandatory = $false)] [string] $appRevision,
+    [Parameter(Position = 5, mandatory = $false)] [string] $managed
 )
+$ErrorActionPreference = "Stop"; $ProgressPreference = "SilentlyContinue"; Set-StrictMode -Version 2.0
```
Comment (freddydk) [inline Actions/BuildPowerPlatform/BuildPowerPlatform.ps1:10]: isn't this set in action.yaml ?

**Review by freddydk — CHANGES_REQUESTED** [review]

File: Actions/BuildPowerPlatform/BuildPowerPlatform.ps1 (hunk 6 above)
Comment (andersgMSFT) [reply Actions/BuildPowerPlatform/BuildPowerPlatform.ps1:10]: Yes, but there was a disconnect from running the tests and running the actions. (i.e. some issue only show up in actions). 

We could also set this in the test if this gives to much duplication.

Comment (freddydk) [reply Actions/BuildPowerPlatform/BuildPowerPlatform.ps1:10]: Yes, it shouldn't be here - it should be set in tests if needed.
Probably in BeforeAll where the .ps1 file is included.

**Review by freddydk — CHANGES_REQUESTED** [review]

**Pull request merged**